File: .claude/hooks/start_pipeline.js
#!/usr/bin/env node
'use strict';

const fs = require('fs');
const path = require('path');

const PROJECT_ROOT = path.resolve(__dirname, '../..');
const STATE_PATH = path.join(PROJECT_ROOT, '.pipeline/pipeline.state');

function main() {
  if (!fs.existsSync(STATE_PATH)) process.exit(0);

  let state;
  try {
    state = JSON.parse(fs.readFileSync(STATE_PATH, 'utf8'));
  } catch {
    process.exit(0);
  }

  if (state.mode === 'pipeline') {
    state.mode = 'free';
    fs.writeFileSync(STATE_PATH, JSON.stringify(state, null, 2));
  }
}

main();

File: .claude/hooks/start_pipeline.py
#!/usr/bin/env python3
"""Start hook: resets pipeline mode to free at session start."""
import json
import sys
from pathlib import Path

PROJECT_ROOT = Path(__file__).resolve().parent.parent.parent  # hooks/ -> .claude/ -> project root
STATE_PATH = PROJECT_ROOT / ".pipeline/pipeline.state"


def main():
    if not STATE_PATH.exists():
        sys.exit(0)
    try:
        state = json.loads(STATE_PATH.read_text())
    except (json.JSONDecodeError, OSError):
        sys.exit(0)
    if state.get("mode") == "pipeline":
        state["mode"] = "free"
        STATE_PATH.write_text(json.dumps(state, indent=2))


if __name__ == "__main__":
    main()
